.github/workflows/test.yml

```diff
@@ -78,7 +78,7 @@ jobs:
         run: cabal check
       - name: Configure
         shell: bash
-        run: cabal configure --enable-tests
+        run: cabal configure --enable-tests --ghc-options='-Wall -Wcompat -Werror'
```

Comment:
I think you only need to enable `-Werror` here, since both `-Wall` and `-Wcompat` are already enabled in the `.cabal` file.